### PR TITLE
translate this page into Chinese: https://kubernetes.io/blog/2022/03/31/ready-for-dockershim-removal/

### PR DESCRIPTION
**Pull request by lishuai-hw**
Translate this page into Chinese: https://kubernetes.io/blog/2022/03/31/ready-for-dockershim-removal/

### CODE REVIEW

**Comment by lishuai-hw** [comment]
#33754

File: content/zh/blog/_posts/2022-03-31-ready-for-dockershim-removal.md [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+
+<!--
+If you are rolling your own cluster or are otherwise unsure whether or not this removal affects you, stay on the safe side and [check to see if you have any dependencies on Docker Engine](/docs/tasks/administer-cluster/migrating-from-dockershim/check-if-dockershim-removal-affects-you/). Please note that using Docker Desktop to build your application containers is not a Docker dependency for your cluster. Container images created by Docker are compliant with the [Open Container Initiative (OCI)](https://opencontainers.org/), a Linux Foundation governance structure that defines industry standards around container formats and runtimes. They will work just fine on any container runtime supported by Kubernetes.
+-->
+如果你正在回滚自己的集群或不确定此删除是否会影响到你，
+请保持安全状态并[检查你对 Docker Engine 是否有依赖](/docs/tasks/administer-cluster/migrating-from-dockershim/check-if-dockershim-removal-affects-you/)。 
+请注意，使用 Docker Desktop 构建应用程序容器并不算是集群对 Docker 有依赖。
+Docker 创建的容器镜像符合 [Open Container Initiative (OCI)](https://opencontainers.org/)，
+这是 Linux 基金会的一种治理架构，它围绕容器格式和运行时定义行业标准。 
+它们可以在 Kubernetes 支持的任何容器运行时上正常工作。
```
Comment (tengqm) [inline content/zh/blog/_posts/2022-03-31-ready-for-dockershim-removal.md:45]:
```suggestion
这些镜像可以在 Kubernetes 支持的任何容器运行时上正常工作。
```

File: content/zh/blog/_posts/2022-03-31-ready-for-dockershim-removal.md [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+如果你使用的是云服务提供商管理的 Kubernetes 服务，
+并且你确定没有更改过容器运行时，那么你可能不需要做任何事情。 
+Amazon EKS、Azure AKS 和 Google GKE 现在都默认使用 containerd，
+但如果你的集群中有任何自定义的节点，你要确保它们不需要被更新。
+要检查节点的运行时，请参考[查明节点上所使用的容器运行时](/zh/docs/tasks/administer-cluster/migrating-from-dockershim/find-out-runtime-you-use/)。
+
+<!--
+Regardless of whether you are rolling your own cluster or using a managed Kubernetes service from a cloud provider, you may need to [migrate telemetry or security agents that rely on Docker Engine](/docs/tasks/administer-cluster/migrating-from-dockershim/migrating-telemetry-and-security-agents/). 
+-->
+无论你是回滚自己的集群还是使用云服务提供商管理的 Kubernetes 服务，
```
Comment (tengqm) [inline content/zh/blog/_posts/2022-03-31-ready-for-dockershim-removal.md:59]:
```suggestion
无论你是在管理自己的集群还是使用云服务提供商管理的 Kubernetes 服务，
```

File: content/zh/blog/_posts/2022-03-31-ready-for-dockershim-removal.md [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+-->
+如果你的 Kubernetes 集群对 Docker Engine 有依赖，
+并且你打算升级到 Kubernetes v1.24 版本（出于安全和类似原因，你最终应该这样做），
+你需要将容器运行时从 Docker Engine 更改为其他方式或使用 [cri-dockerd](https://github.com/Mirantis/cri-dockerd)。 
+由于 [containerd](https://containerd.io/) 是一个已经毕业的 CNCF 项目，
+并且是 Docker 本身的运行时，因此用它作为容器运行时的替代方式是一个安全的选择。
+幸运的是，Kubernetes 项目已经以 containerd 为例，
+记录了[更改节点容器运行时](/zh/docs/tasks/administer-cluster/migrating-from-dockershim/change-runtime-containerd/)的过程。
```
Comment (tengqm) [inline content/zh/blog/_posts/2022-03-31-ready-for-dockershim-removal.md:76]:
```suggestion
提供了[更改节点容器运行时](/zh/docs/tasks/administer-cluster/migrating-from-dockershim/change-runtime-containerd/)的过程文档。
```

File: content/zh/blog/_posts/2022-03-31-ready-for-dockershim-removal.md [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+切换到其它支持的运行时的说明与此类似。
```
Comment (tengqm) [inline content/zh/blog/_posts/2022-03-31-ready-for-dockershim-removal.md:77]:
```suggestion
切换到其它支持的运行时的操作指令与此类似。
```

File: content/zh/blog/_posts/2022-03-31-ready-for-dockershim-removal.md [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+
+<!--
+## I want to upgrade Kubernetes, and I need to maintain compatibility with Docker as a runtime. What are my options?
+-->
+## 我想升级 Kubernetes，并且我需要保持与 Docker 作为运行时的兼容性。我有哪些选择？
+
+<!--
+Fear not, you aren’t being left out in the cold and you don’t have to take the security risk of staying on an old version of Kubernetes. Mirantis and Docker have jointly released, and are maintaining, a replacement for dockershim. That replacement is called [cri-dockerd](https://github.com/Mirantis/cri-dockerd). If you do need to maintain compatibility with Docker as a runtime, install cri-dockerd following the instructions in the project’s documentation.
+-->
+别担心，你不会被冷落，也不必冒着安全风险继续使用旧版本的 Kubernetes 。
```
Comment (tengqm) [inline content/zh/blog/_posts/2022-03-31-ready-for-dockershim-removal.md:87]:
```suggestion
别担心，你不会被冷落，也不必冒着安全风险继续使用旧版本的 Kubernetes。 
```

File: content/zh/blog/_posts/2022-03-31-ready-for-dockershim-removal.md [hunk 6]
```diff
@@ -0,0 +1,111 @@
...
+
+
+<!--
+Yes. As long as you go into this release aware of the changes being made and the details of your own clusters, and you make sure to communicate clearly with your development teams, it will be minimally dramatic. You may have some changes to make to your cluster, application code, or scripts, but all of these requirements are documented. Switching from using Docker Engine as your runtime to using [one of the other supported container runtimes](/docs/setup/production-environment/container-runtimes/) effectively means removing the middleman, since the purpose of dockershim is to access the container runtime used by Docker itself. From a practical perspective, this removal is better both for you and for Kubernetes maintainers in the long-run.
+-->
+是的。只要你深入了解此版本所做的变更和你自己集群的详细信息，
+并确保与你的开发团队进行清晰的沟通，它的不确定性就会降到最低。 
+你可能需要对集群、应用程序代码或脚本进行一些更改，但所有这些要求都已经有说明指导。 
+从使用 Docker Engine 作为运行时，切换到使用[其他任何一种支持的容器运行时](/zh/docs/setup/production-environment/container-runtimes/)，
+这意味着移除中间过程，因为 dockershim 的目的是访问 Docker 本身使用的容器运行时。
```
Comment (tengqm) [inline content/zh/blog/_posts/2022-03-31-ready-for-dockershim-removal.md:105]:
```suggestion
这意味着移除了中间层的组件，因为 dockershim 的作用是访问 Docker 本身使用的容器运行时。 
```

File: content/zh/blog/_posts/2022-03-31-ready-for-dockershim-removal.md [hunk 7]
```diff
@@ -0,0 +1,111 @@
...
+<!--
+## First, does this even affect you?
+-->
+## 首先，这对你有影响吗？
+
+<!--
+If you are rolling your own cluster or are otherwise unsure whether or not this removal affects you, stay on the safe side and [check to see if you have any dependencies on Docker Engine](/docs/tasks/administer-cluster/migrating-from-dockershim/check-if-dockershim-removal-affects-you/). Please note that using Docker Desktop to build your application containers is not a Docker dependency for your cluster. Container images created by Docker are compliant with the [Open Container Initiative (OCI)](https://opencontainers.org/), a Linux Foundation governance structure that defines industry standards around container formats and runtimes. They will work just fine on any container runtime supported by Kubernetes.
+-->
+如果你正在回滚自己的集群或不确定此删除是否会影响到你，
+请保持安全状态并[检查你对 Docker Engine 是否有依赖](/docs/tasks/administer-cluster/migrating-from-dockershim/check-if-dockershim-removal-affects-you/)。
```
[the remark this answers is not in the document]
Comment (lishuai-hw) [reply content/zh/blog/_posts/2022-03-31-ready-for-dockershim-removal.md:41]: @tengqm 
Thanks for your review. This page (/docs/tasks/administer-cluster/migrating-from-dockershim/check-if-dockershim-removal-affects-you/) doesn't have a Chinese translation doc. Should I use /zh/docs/..... or without /zh ?

[the remark this answers is not in the document]
Comment (lishuai-hw) [reply content/zh/blog/_posts/2022-03-31-ready-for-dockershim-removal.md]: I will submit a new ISSUE about this page (/docs/tasks/administer-cluster/migrating-from-dockershim/check-if-dockershim-removal-affects-you/), and translate it to Chinese.
This conversation is accept, and I set it to resolved.
Thank you.

**Comment by lishuai-hw** [comment]
@tengqm 
Thanks for your review. I have modified my zh file according to your suggestion. Pls check again. Thank you.

**Comment by tengqm** [comment]
Please squash the PR into one commit.

File: content/zh/blog/_posts/2022-03-31-ready-for-dockershim-removal.md (hunk 7 above)
[the remark this answers is not in the document]
Comment (tengqm) [reply content/zh/blog/_posts/2022-03-31-ready-for-dockershim-removal.md]: OK. Keep the `/zh` prefix and file an issue then.

**Comment by lishuai-hw** [comment]
> 



> Please squash the PR into one commit.

@tengqm hi master, I squashed the last two commits.

File: content/zh/blog/_posts/2022-03-31-ready-for-dockershim-removal.md (hunk 7 above)
[the remark this answers is not in the document]
Comment (lishuai-hw) [reply content/zh/blog/_posts/2022-03-31-ready-for-dockershim-removal.md]: I find that this Chinese page has already exist, but its name(check-if-dockershim-**deprecation**-affects-you.md) is inconsistent with its English name(check-if-dockershim-removal-affects-you.md). I submit  a ISSUE about this #33790

**Pull request merged**